bridges/BakaUpdatesMangaReleasesBridge.php

```diff
@@ -0,0 +1,84 @@
+<?php
+class BakaUpdatesMangaReleasesBridge extends BridgeAbstract {
+	const NAME = 'Baka Updates Manga Releases';
+	const URI = 'https://www.mangaupdates.com/';
+	const DESCRIPTION = 'Get the latest series releases';
+	const MAINTAINER = 'fulmeek';
+	const PARAMETERS = array(array(
+		'series_id' => array(
+			'name'			=> 'Series ID',
+			'exampleValue'	=> '12345'
+		)
+	));
+	const LIMIT_ITEMS = 10;
+
+	private $feedName = '';
+
+	public function collectData() {
+		$series_id = preg_replace('/[^0-9]/', '', $this->getInput('series_id'));
+		if (empty($series_id))
+			returnServerError('Invalid series ID');
+
+		$html = getSimpleHTMLDOM(self::URI . 'releases.html?search=' . $series_id . '&stype=series')
+			or returnServerError('Series not found');
+
+		$objTitle = $html->find('td[class="text pad"]', 1);
+		if ($objTitle)
+			$this->feedName = $objTitle->plaintext;
+
+		$itemlist = $html->find('td#main_content table table table tr');
+		if (!$itemlist)
+			returnServerError('No releases');
+
+		$limit = self::LIMIT_ITEMS;
+		foreach($itemlist as $element) {
+			if (!$element->find('td[class="text pad"]', 0))
+				continue;
+			if ($limit <= 0)
+				break;
+
+			$item = array();
+			$title = array();
+
+			$item['content'] = '';
+
+			$objDate = $element->find('td[class="text pad"]', 0);
+			if ($objDate)
+				$item['timestamp'] = strtotime($objDate->plaintext);
+
+			$objTitle = $element->find('td[class="text pad"]', 1);
+			if ($objTitle) {
+				$title[] = html_entity_decode($objTitle->plaintext);
+				$item['content'] .= '<p>Series: ' . $objTitle->innertext . '</p>';
+			}
+
+			$objVolume = $element->find('td[class="text pad"]', 2);
+			if ($objVolume && !empty($objVolume->plaintext))
+				$title[] = 'Vol.' . $objVolume->plaintext;
+
+			$objChapter = $element->find('td[class="text pad"]', 3);
+			if ($objChapter && !empty($objChapter->plaintext))
+				$title[] = 'Chp.' . $objChapter->plaintext;
+
+			$objAuthor = $element->find('td[class="text pad"]', 4);
+			if ($objAuthor && !empty($objAuthor->plaintext)) {
+				$item['author'] = html_entity_decode($objAuthor->plaintext);
+				$item['content'] .= '<p>Groups: ' . $objAuthor->innertext . '</p>';
+			}
+
+			$item['title']	= implode(' ', $title);
+			$item['uri']	= 'urn:sha1:' . hash('sha1', $item['title']);
+
+			$this->items[] = $item;
+
+			--$limit;
```

Comment:
Instead of decreasing the limit, just check the size of the items array:

```suggestion
			if(count($this->items) >= $limit) {
				break;
			}
```

Reply:
See above, I can't precisely fetch the desired rows, so counting would not work here. If it would, I wouldn't need this at all. I have to limit the items, otherwise there could be easily 100 ones.

Reply:
This is about breaking the loop when the desired number of elements were added to the feed (i.e. only return 10 items in the feed).